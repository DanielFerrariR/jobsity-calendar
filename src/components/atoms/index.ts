export { default as Box } from './box'
export { default as Button } from './button'
export { default as Paper } from './paper'
export { default as Typography } from './typography'
export { default as TextField } from './text_field'
export { default as Container } from './container'
export { default as AppBar } from './appbar'
export { default as IconButton } from './icon_button'
export { default as Toolbar } from './toolbar'
export { default as Image } from './image'
export { default as Grid } from './grid'
export { default as Link } from './link'
export { default as Checkbox } from './checkbox'
export { default as Accordion } from './accordion'
export { default as AccordionSummary } from './accordion_summary'
export { default as AccordionDetails } from './accordion_details'
export { default as Tooltip } from './tooltip'
export { default as Modal } from './modal'
export { default as MenuItem } from './menu_item'
export { default as ListItemIcon } from './list_item_icon'
export { default as ListItemText } from './list_item_text'
export { default as ListItemSecondaryAction } from './list_item_secondary_action'
export { default as MenuList } from './menu_list'
export { default as Popover } from './popover'
export { default as Fade } from './fade'
export { default as Backdrop } from './backdrop'
export { default as CircularProgress } from './circular_progress'
export { default as LinearProgress } from './linear_progress'
export { default as Skeleton } from './skeleton'
export { default as List } from './list'
export { default as ListItem } from './list_item'
export { default as ClickAwayListener } from './click_away_listener'
export { default as Badge } from './badge'
export { default as Portal } from './portal'
export { default as Snackbar } from './snackbar'
export { default as Step } from './step'
export { default as StepContent } from './stepContent'
export { default as StepLabel } from './stepLabel'
export { default as Stepper } from './stepper'
export { default as TableBody } from './table_body'
export { default as TableContainer } from './table_container'
export { default as TableHead } from './table_head'
export { default as TableRow } from './table_row'
export { default as Table } from './table'
export { default as TableCell } from './table_cell'
export { default as DateTimePicker } from './date_time_picker'
