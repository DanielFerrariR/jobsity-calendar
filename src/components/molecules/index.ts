export { CustomTableHead, CustomTableBody } from './custom_table'
export { default as Modal } from './modal'
export { default as ReminderCard } from './reminder_card'
