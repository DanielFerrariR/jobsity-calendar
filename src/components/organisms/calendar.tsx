import React, { useState } from 'react'
import {
  Tooltip,
  Paper,
  Typography,
  Button,
  IconButton,
  Box,
  TableContainer,
  Table,
  TableCell,
  TableRow
} from 'src/components/atoms'
import {
  CustomTableHead,
  CustomTableBody,
  ReminderCard
} from 'src/components/molecules'
import { ArrowForwardIos } from '@material-ui/icons'
import { useTheme, useMediaQuery } from '@material-ui/core'
import ListRemindersModal from './list_reminders_modal'
import CreateReminderModal from './create_reminder_modal'

const Calendar: React.FC = () => {
  const [openCreateReminderModal, setOpenCreateReminderModal] = useState(false)
  const [openListRemindersModal, setOpenListRemindersModal] = useState<
    false | Date
  >(false)
  const theme = useTheme()
  const xsAndDown = useMediaQuery(theme.breakpoints.down('xs'))
  const today =
    process.env.NODE_ENV === 'test' || window.Cypress
      ? new Date('2020/10/20')
      : new Date()
  const [year, setYear] = useState(today.getFullYear())
  const [month, setMonth] = useState(today.getMonth())
  const monthNames = [
    'January',
    'February',
    'March',
    'April',
    'May',
    'June',
    'July',
    'August',
    'September',
    'October',
    'November',
    'December'
  ]
  const weekDaysNames = [
    'Sunday',
    'Monday',
    'Tuesday',
    'Wednesday',
    'Thursday',
    'Friday',
    'Saturday'
  ]

  const setToday = () => {
    setMonth(today.getMonth())
    setYear(today.getFullYear())
  }

  const setPreviousMonth = () => {
    if (month > 0) {
      setMonth(month - 1)
    } else {
      setYear(year - 1)
      setMonth(11)
    }
  }

  const setNextMonth = () => {
    if (month < 11) {
      setMonth(month + 1)
    } else {
      setYear(year + 1)
      setMonth(0)
    }
  }

  const createBodyLine = (current: number) => {
    const firstDay = new Date(year, month).getDay()
    const previousMonth = month === 0 ? 11 : month - 1
    const daysInPastMonth = 32 - new Date(year, previousMonth, 32).getDate()
    const daysInMonth = 32 - new Date(year, month, 32).getDate()
    const bodyLine = []

    for (let count = current * 7; count < current * 7 + 7; count += 1) {
      if (count < firstDay) {
        const newDay = daysInPastMonth + count - firstDay + 1
        const newMonth = month > 0 ? month - 1 : 11
        const newYear = month > 0 ? year : year - 1
        const newDate = new Date(newYear, newMonth, newDay)
        bodyLine.push(
          <TableCell
            key={count}
            onClick={() => setOpenListRemindersModal(newDate)}
            data-testid={`calendar-table-cell-past-${newDay}`}
          >
            <Typography fontWeight="500" color={theme.palette.grey[500]} mb={1}>
              {newDay}
            </Typography>
            <ReminderCard
              date={newDate}
              testId={`reminder-card-past-${newDay}`}
            />
          </TableCell>
        )
      } else if (count - firstDay + 1 > daysInMonth) {
        const newDay = count - daysInMonth - firstDay + 1
        const newMonth = month < 11 ? month + 1 : 0
        const newYear = month < 11 ? year : year + 1
        const newDate = new Date(newYear, newMonth, newDay)

        bodyLine.push(
          <TableCell
            key={count}
            onClick={() => setOpenListRemindersModal(newDate)}
            data-testid={`calendar-table-cell-future-${newDay}`}
          >
            <Typography fontWeight="500" color={theme.palette.grey[500]} mb={1}>
              {newDay}
            </Typography>
            <ReminderCard
              date={newDate}
              testId={`reminder-card-future-${newDay}`}
            />
          </TableCell>
        )
      } else {
        const newDay = count - firstDay + 1
        const isWeekend = count === current * 7 || count === current * 7 + 6
        const isToday =
          count - firstDay + 1 === today.getDate() &&
          month === today.getMonth() &&
          year === today.getFullYear()
        let dayColor = ''
        const newDate = new Date(year, month, newDay)

        if (isToday) {
          dayColor = theme.palette.error.main
        } else if (isWeekend) {
          dayColor = theme.palette.primary.main
        }

        bodyLine.push(
          <TableCell
            key={count}
            onClick={() => setOpenListRemindersModal(newDate)}
            data-testid={`calendar-table-cell-current-${newDay}`}
          >
            <Typography fontWeight="500" color={dayColor} mb={1}>
              {newDay}
            </Typography>
            <ReminderCard
              date={newDate}
              testId={`reminder-card-current-${newDay}`}
            />
          </TableCell>
        )
      }
    }

    return bodyLine
  }

  const createBody = () => {
    const body = []

    for (let count = 0; count < 6; count += 1) {
      body.push(<TableRow key={count}>{createBodyLine(count)}</TableRow>)
    }

    return body
  }

  return (
    <>
      <Paper p={2}>
        <Box
          display="flex"
          alignItems="center"
          mb={4}
          flexDirection={xsAndDown && 'column'}
          position="relative"
        >
          <Box
            display="flex"
            alignItems="center"
            position={!xsAndDown && 'absolute'}
            mb={xsAndDown && 2}
            width={1}
            justifyContent="space-between"
            flexDirection={xsAndDown && 'column'}
          >
            <Box mb={xsAndDown && 2}>
              <Tooltip title="Back">
                <IconButton
                  aria-label="Back"
                  onClick={setPreviousMonth}
                  disabled={month === 0 && year === 1970}
                  data-testid="calendar-button-back"
                >
                  <ArrowForwardIos style={{ transform: 'rotate(180deg)' }} />
                </IconButton>
              </Tooltip>
              <Tooltip title="Forward">
                <IconButton
                  aria-label="Forward"
                  onClick={setNextMonth}
                  mr={1}
                  disabled={month === 11 && year === 9999}
                  data-testid="calendar-button-forward"
                >
                  <ArrowForwardIos />
                </IconButton>
              </Tooltip>
              <Button
                onClick={setToday}
                data-testid="calendar-button-set-today"
              >
                Today
              </Button>
            </Box>
            <Button
              ml={1}
              onClick={() => setOpenCreateReminderModal(true)}
              data-testid="calendar-button-create-reminder"
            >
              Create reminder
            </Button>
          </Box>
          <Typography
            variant="h6"
            textAlign="center"
            width={1}
            ml={1}
            data-testid="calendar-typography-date"
          >
            {monthNames[month]} {year}
          </Typography>
        </Box>
        <TableContainer>
          <Table style={{ minWidth: 1200 }}>
            <CustomTableHead>
              <TableRow>
                {weekDaysNames.map((each, index) => {
                  return (
                    <TableCell key={index}>
                      <Typography textAlign="center" fontWeight="500">
                        {each}
                      </Typography>
                    </TableCell>
                  )
                })}
              </TableRow>
            </CustomTableHead>
            <CustomTableBody>{createBody()}</CustomTableBody>
          </Table>
        </TableContainer>
      </Paper>
      <ListRemindersModal
        open={openListRemindersModal}
        setOpen={setOpenListRemindersModal}
      />
      <CreateReminderModal
        open={openCreateReminderModal}
        setOpen={setOpenCreateReminderModal}
      />
    </>
  )
}

export default Calendar
