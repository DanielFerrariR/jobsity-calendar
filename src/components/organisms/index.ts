export { default as Calendar } from './calendar'
export { default as ListRemindersModal } from './list_reminders_modal'
export { default as CreateReminderModal } from './create_reminder_modal'
export { default as EditReminderModal } from './edit_reminder_modal'
