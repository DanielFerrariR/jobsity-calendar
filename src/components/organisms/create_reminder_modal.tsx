import React, { Dispatch, SetStateAction, useState } from 'react'
import {
  Box,
  Typography,
  Button,
  TextField,
  DateTimePicker,
  CircularProgress
} from 'src/components/atoms'
import { Modal } from 'src/components/molecules'
import { useDispatch, useSelector } from 'src/store'
import { createReminder } from 'src/store/reminders'

interface Props {
  open: boolean
  setOpen: Dispatch<SetStateAction<boolean>>
}

interface Form {
  text: string
  date: null | Date
  city: string
  color: string
}

const CreateReminderModal: React.FC<Props> = ({ open, setOpen }) => {
  const [loading, setLoading] = useState(false)
  const reminders = useSelector((state) => state.reminders)
  const dispatch = useDispatch()
  const colors = [
    '#f44336',
    '#2196f3',
    '#4caf50',
    '#9c27b0',
    '#ff9800',
    '#795548'
  ]
  const [form, setForm] = useState<Form>({
    text: '',
    date: new Date(),
    city: '',
    color: colors[0]
  })

  const handleChange = (name: string) => (
    event: React.ChangeEvent<HTMLInputElement>
  ) => {
    setForm({
      ...form,
      [name]: event.target.value
    })
  }

  const handleFormSubmit = (event: React.FormEvent) => {
    event.preventDefault()

    if (!form.date) {
      return
    }

    setLoading(true)

    dispatch(
      createReminder(reminders || [], {
        text: form.text,
        date: form.date,
        city: form.city,
        color: form.color
      })
    )

    setLoading(false)
    setOpen(false)
  }

  return (
    <Modal open={open} setOpen={setOpen}>
      <Box component="form" onSubmit={handleFormSubmit}>
        <Typography mb={3} variant="h6">
          Create Reminder
        </Typography>
        <Typography mb={1}>Choose a color</Typography>
        <Box display="flex" alignItems="center" mb={3}>
          {colors.map((each) => {
            return (
              <Box
                key={each}
                style={{ cursor: 'pointer' }}
                borderRadius={4}
                width={24}
                height={24}
                bgcolor={each}
                border={form.color === each && '3px solid black'}
                onClick={() => setForm({ ...form, color: each })}
                mr={1}
              />
            )
          })}
        </Box>
        <TextField
          label="Text"
          inputProps={{
            maxLength: 30
          }}
          mb={3}
          width={1}
          onChange={handleChange('text')}
          required
        />
        <DateTimePicker
          label="Date"
          inputVariant="outlined"
          value={form.date}
          onChange={(date) => setForm({ ...form, date })}
          style={{ marginBottom: '24px' }}
          fullWidth
        />
        <TextField
          label="City"
          width={1}
          mb={3}
          onChange={handleChange('city')}
          required
        />
        <Box position="relative">
          <Button width={1} mb={3} type="submit" disabled={loading}>
            Create
          </Button>
          {loading && (
            <CircularProgress
              color="primary.main"
              size={24}
              position="absolute"
              top="calc(50% - 12px)"
              left="calc(50% - 12px)"
            />
          )}
        </Box>
        <Button width={1} variant="outlined" onClick={() => setOpen(false)}>
          Close
        </Button>
      </Box>
    </Modal>
  )
}

export default CreateReminderModal
