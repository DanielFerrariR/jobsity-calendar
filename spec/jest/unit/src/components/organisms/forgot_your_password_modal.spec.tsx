import React from 'react'
import { ForgotYourPasswordModal } from 'src/components/organisms'
import { render } from 'spec/jest/config/test_utils'
import { axe } from 'jest-axe'

describe('testing ForgotYourPasswordModal component', () => {
  beforeEach(() => {
    jest.useFakeTimers()
  })

  afterEach(() => {
    jest.useRealTimers()
  })

  test('should snapshot works with english language', () => {
    const { container } = render(
      <ForgotYourPasswordModal open setOpen={(): void => {}} />
    )

    expect(container).toMatchSnapshot()
  })

  test('should snapshot works with portuguese language', () => {
    const { container } = render(
      <ForgotYourPasswordModal open setOpen={(): void => {}} />,
      {
        language: 'pt'
      }
    )

    expect(container).toMatchSnapshot()
  })

  test('should snapshot works with spanish language', () => {
    const { container } = render(
      <ForgotYourPasswordModal open setOpen={(): void => {}} />,
      {
        language: 'es'
      }
    )

    expect(container).toMatchSnapshot()
  })

  test('should the page be accessible', async () => {
    const { container } = render(
      <ForgotYourPasswordModal open setOpen={(): void => {}} />
    )

    jest.useRealTimers()

    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
