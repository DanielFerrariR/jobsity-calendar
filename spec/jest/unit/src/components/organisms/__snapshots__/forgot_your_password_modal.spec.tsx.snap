// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing ForgotYourPasswordModal component should snapshot works with english language 1`] = `
<div
  aria-hidden="true"
>
  <div
    aria-describedby="description"
    aria-labelledby="modal-title"
    class="ForwardRef(Modal)-root ForwardRef(Modal)-root"
    data-testid="modal-alert"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root WithStyles(ForwardRef(Backdrop))-root"
      style="opacity: 1; webkit-transition: opacity 500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiPaper-root Component-root Component-root ForwardRef(Fade)-root MuiPaper-elevation5 MuiPaper-rounded"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <form
        class="MuiBox-root MuiBox-root Component-root Component-root"
      >
        <h6
          class="MuiTypography-root Component-root Component-root MuiTypography-h6 MuiTypography-alignCenter"
        >
          Resetting password
        </h6>
        <p
          class="MuiTypography-root Component-root Component-root MuiTypography-body1 MuiTypography-alignCenter"
        >
          In order to set a new password you need to enter your subscribed e-mail and the token you received in your e-mail when you have created your account.
        </p>
        <div
          class="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root WithStyles(ForwardRef(TextField))-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-required MuiInputLabel-required"
            data-shrink="true"
            for="forgot-your-password-modal-email"
            id="forgot-your-password-modal-email-label"
          >
            E-mail
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input"
              id="forgot-your-password-modal-email"
              maxlength="100"
              placeholder="Type your subscribed e-mail"
              required=""
              type="email"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled PrivateNotchedOutline-legendNotched"
              >
                <span>
                  E-mail
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root WithStyles(ForwardRef(TextField))-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-required MuiInputLabel-required"
            data-shrink="true"
            for="forgot-your-password-modal-token"
            id="forgot-your-password-modal-token-label"
          >
            Token
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input"
              id="forgot-your-password-modal-token"
              maxlength="100"
              placeholder="Type your token"
              required=""
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled PrivateNotchedOutline-legendNotched"
              >
                <span>
                  Token
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiGrid-root WithStyles(ForwardRef(Grid))-root WithStyles(ForwardRef(Grid))-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root WithStyles(ForwardRef(Grid))-root WithStyles(ForwardRef(Grid))-root MuiGrid-item MuiGrid-grid-xs-6"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined Component-root Component-root MuiButton-outlinedSecondary MuiButton-outlinedSizeLarge MuiButton-sizeLarge"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Cancel
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root WithStyles(ForwardRef(Grid))-root WithStyles(ForwardRef(Grid))-root MuiGrid-item MuiGrid-grid-xs-6"
          >
            <div
              class="MuiBox-root MuiBox-root Component-root Component-root"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained Component-root Component-root MuiButton-containedSecondary MuiButton-containedSizeLarge MuiButton-sizeLarge"
                tabindex="0"
                type="submit"
              >
                <span
                  class="MuiButton-label"
                >
                  Validate
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</div>
`;

exports[`testing ForgotYourPasswordModal component should snapshot works with portuguese language 1`] = `
<div
  aria-hidden="true"
>
  <div
    aria-describedby="description"
    aria-labelledby="modal-title"
    class="ForwardRef(Modal)-root ForwardRef(Modal)-root"
    data-testid="modal-alert"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root WithStyles(ForwardRef(Backdrop))-root"
      style="opacity: 1; webkit-transition: opacity 500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiPaper-root Component-root Component-root ForwardRef(Fade)-root MuiPaper-elevation5 MuiPaper-rounded"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <form
        class="MuiBox-root MuiBox-root Component-root Component-root"
      >
        <h6
          class="MuiTypography-root Component-root Component-root MuiTypography-h6 MuiTypography-alignCenter"
        >
          Redefinir senha
        </h6>
        <p
          class="MuiTypography-root Component-root Component-root MuiTypography-body1 MuiTypography-alignCenter"
        >
          Para definir uma nova senha, você precisa digitar o seu email e o token que recebeu no seu email quando criou sua conta.
        </p>
        <div
          class="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root WithStyles(ForwardRef(TextField))-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-required MuiInputLabel-required"
            data-shrink="true"
            for="forgot-your-password-modal-email"
            id="forgot-your-password-modal-email-label"
          >
            E-mail
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input"
              id="forgot-your-password-modal-email"
              maxlength="100"
              placeholder="Digite o e-mail cadastrado"
              required=""
              type="email"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled PrivateNotchedOutline-legendNotched"
              >
                <span>
                  E-mail
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root WithStyles(ForwardRef(TextField))-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-required MuiInputLabel-required"
            data-shrink="true"
            for="forgot-your-password-modal-token"
            id="forgot-your-password-modal-token-label"
          >
            Token
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input"
              id="forgot-your-password-modal-token"
              maxlength="100"
              placeholder="Digite seu token"
              required=""
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled PrivateNotchedOutline-legendNotched"
              >
                <span>
                  Token
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiGrid-root WithStyles(ForwardRef(Grid))-root WithStyles(ForwardRef(Grid))-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root WithStyles(ForwardRef(Grid))-root WithStyles(ForwardRef(Grid))-root MuiGrid-item MuiGrid-grid-xs-6"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined Component-root Component-root MuiButton-outlinedSecondary MuiButton-outlinedSizeLarge MuiButton-sizeLarge"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Cancelar
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root WithStyles(ForwardRef(Grid))-root WithStyles(ForwardRef(Grid))-root MuiGrid-item MuiGrid-grid-xs-6"
          >
            <div
              class="MuiBox-root MuiBox-root Component-root Component-root"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained Component-root Component-root MuiButton-containedSecondary MuiButton-containedSizeLarge MuiButton-sizeLarge"
                tabindex="0"
                type="submit"
              >
                <span
                  class="MuiButton-label"
                >
                  Validar
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</div>
`;

exports[`testing ForgotYourPasswordModal component should snapshot works with spanish language 1`] = `
<div
  aria-hidden="true"
>
  <div
    aria-describedby="description"
    aria-labelledby="modal-title"
    class="ForwardRef(Modal)-root ForwardRef(Modal)-root"
    data-testid="modal-alert"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root WithStyles(ForwardRef(Backdrop))-root"
      style="opacity: 1; webkit-transition: opacity 500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiPaper-root Component-root Component-root ForwardRef(Fade)-root MuiPaper-elevation5 MuiPaper-rounded"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <form
        class="MuiBox-root MuiBox-root Component-root Component-root"
      >
        <h6
          class="MuiTypography-root Component-root Component-root MuiTypography-h6 MuiTypography-alignCenter"
        >
          Restablecer contraseña
        </h6>
        <p
          class="MuiTypography-root Component-root Component-root MuiTypography-body1 MuiTypography-alignCenter"
        >
          Para establecer una nueva contraseña, debe ingresar su correo electrónico y el token que recibió en su correo electrónico cuando creó su cuenta.
        </p>
        <div
          class="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root WithStyles(ForwardRef(TextField))-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-required MuiInputLabel-required"
            data-shrink="true"
            for="forgot-your-password-modal-email"
            id="forgot-your-password-modal-email-label"
          >
            E-mail
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input"
              id="forgot-your-password-modal-email"
              maxlength="100"
              placeholder="Ingrese el e-mail registrado"
              required=""
              type="email"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled PrivateNotchedOutline-legendNotched"
              >
                <span>
                  E-mail
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root WithStyles(ForwardRef(TextField))-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-required MuiInputLabel-required"
            data-shrink="true"
            for="forgot-your-password-modal-token"
            id="forgot-your-password-modal-token-label"
          >
            Token
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input"
              id="forgot-your-password-modal-token"
              maxlength="100"
              placeholder="Ingrese su token"
              required=""
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled PrivateNotchedOutline-legendNotched"
              >
                <span>
                  Token
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiGrid-root WithStyles(ForwardRef(Grid))-root WithStyles(ForwardRef(Grid))-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root WithStyles(ForwardRef(Grid))-root WithStyles(ForwardRef(Grid))-root MuiGrid-item MuiGrid-grid-xs-6"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined Component-root Component-root MuiButton-outlinedSecondary MuiButton-outlinedSizeLarge MuiButton-sizeLarge"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Cancelar
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root WithStyles(ForwardRef(Grid))-root WithStyles(ForwardRef(Grid))-root MuiGrid-item MuiGrid-grid-xs-6"
          >
            <div
              class="MuiBox-root MuiBox-root Component-root Component-root"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained Component-root Component-root MuiButton-containedSecondary MuiButton-containedSizeLarge MuiButton-sizeLarge"
                tabindex="0"
                type="submit"
              >
                <span
                  class="MuiButton-label"
                >
                  Validar
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</div>
`;
